site/public/js/user-profile.js

```diff
@@ -235,11 +235,15 @@ function updateUserPreferredNames() {
                     displaySuccessMessage(data.message);
                     // update the preferred names
                     const icon = '<i class="fas fa-pencil-alt"></i>';
-                    $('#givenname-row .icon').html(`${icon} ${data.given_name}`);
-                    $('#familyname-row .icon').html(`${icon} ${data.family_name}`);
+                    $('#givenname-row .icon').html(`${icon} `).append($('<span>').text(data.displayed_given_name));
+                    $('#familyname-row .icon').html(`${icon} `).append($('<span>').text(data.displayed_family_name));
```

Comment:
I still don't like using `.html()` here, but I suppose it's not worth blocking this PR on it...